File: plus-one.js
// You are given a large integer represented as an integer array digits, where each 
// digits[i] is the ith digit of the integer. The digits are ordered from most significant to least 
// significant in left-to-right order. The large integer does not contain any leading 0's.
// Increment the large integer by one and return the resulting array of digits.

// /**
//  * @param {number[]} digits
//  * @return {number[]}
//  **/
var plusOne = function(digits) {
  let number = BigInt(digits.join("")) + BigInt(1);
  let arr = Array.from(String(number), Number);
  return arr;  
};

plusOne([1,2,3])

// [1,2,4]

File: plus-one.rb
# @param {Integer[]} digits
# @return {Integer[]}

def plus_one(digits)
    nums = digits.join()
    arr_strings = (Integer(nums) + 1).to_s.split(//)
    result = []
    arr_strings.each do |i|
     result.push(Integer(i)) 
        
    end
    print(result)
    return result
end

plus_one([9])